ionoscloud/resource_server_vcpu.go

```diff
@@ -0,0 +1,355 @@
+							Computed: true,
+						},
+						"boot_server": {
+							Type:        schema.TypeString,
+							Description: "The UUID of the attached server.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"nic": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"mac": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"lan": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"dhcp": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"ips": {
+							Type: schema.TypeList,
+							Elem: &schema.Schema{
+								Type:             schema.TypeString,
+								DiffSuppressFunc: utils.DiffEmptyIps,
+							},
+							Description: "Collection of IP addresses assigned to a nic. Explicitly assigned public IPs need to come from reserved IP blocks, Passing value null or empty array will assign an IP address automatically.",
+							Computed:    true,
+							Optional:    true,
+						},
+						"firewall_active": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"firewall_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"device_number": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"pci_slot": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"firewall": {
+							Description: "Firewall rules created in the server resource. The rules can also be created as separate resources outside the server resource",
+							Type:        schema.TypeList,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"protocol": {
+										Type:             schema.TypeString,
+										Required:         true,
+										DiffSuppressFunc: utils.DiffToLower,
+										ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+									},
+									"source_mac": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"source_ip": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"target_ip": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"port_range_start": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ValidateDiagFunc: validation.ToDiagFunc(func(v interface{}, k string) (ws []string, errors []error) {
+											if v.(int) < 1 && v.(int) > 65534 {
```

Comment:
I think this can be replaced with IntBetween validation fund

Reply:
@cristiGuranIonos done in 35a13f8951cffce1c4c25e74f6242197e6cf8354